File: cs441-customDrawing/cs441-customDrawing/ViewController.m
//
//  ViewController.m
//  cs441-customDrawing
//
//  Created by Theresa Gundel on 3/1/21.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController
@synthesize link;
@synthesize drawSome;

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    //creates an obj connected to OS, which knows what to draw to screen. this obj sends a message. vc class called by display link. call the method tick.
    link = [CADisplayLink displayLinkWithTarget:self selector:@selector(tick:)];
    //OS has a main loop. we are adding this link to the loop for the OS to do.
    [link addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSDefaultRunLoopMode];
    //5 frames per second so lower CPU
    [link setPreferredFramesPerSecond:5];
    [link setPaused:NO];
}

-(void) tick:(CADisplayLink *)sender
{
    [drawSome animate];
}

@end

File: cs441-customDrawing/cs441-customDrawing/ViewController.swift
//
//  ViewController.swift
//  cs441-customDrawing
//
//  Created by Theresa Gundel on 3/3/21.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var fishBowl: fishBowl!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        let displayLink = CADisplayLink(target: self, selector: #selector(update))
        displayLink.add(to: RunLoop.main, forMode: RunLoop.Mode.default)
    }
    
    @objc func update(){
        fishBowl.update()
    }

}

